lib/puppet/type/netapp_qtree.rb

```diff
@@ -8,23 +8,54 @@
   ensurable
 
   newparam(:name) do
-    desc "The qtree name."
     isnamevar
+    desc "The qtree name."
     validate do |value|
-      unless value =~ /^[\w\-]+$/
+      unless (value =~ /^[\w\-\.]+$/) or (value =~ /^\/\w+\/[\w\-\.]+$/)
          raise ArgumentError, "%s is not a valid qtree name." % value
       end
     end
   end
 
   newparam(:volume) do
     desc "The volume to create qtree against."
+    defaultto do
+      if @resource[:name].count('/') == 2
+        @resource[:name].split('/')[1] 
+      else
+        ""
+      end
+    end
```

Comment:
Hi hunner, I'm not a ruby programmer expert and just a system admin so I'm not sure to have understood well what exactly I have to do. My coding is probably not the best one :) I'm not so expert also with git... Should I do the changes and add the new commit to the MR with those ?

If I've understood well I should : 
- put this title_patterns in the  netatt_qtree.rb
- remove the defaultto part from :qtname and :volume 
- Should I put the :namevar => true to both :qtname and :volume ?
What about :name ? In the example the 2 namevar are the param used in the title_patterns. Should I remove it completely?

All those checks and defaults was done so that all will still continue to work as it was before but also being able to call the netapp_qtree passing all the parameters (so if passing qtname and volume it use them instead of the titlevar.
So old behaviour (in which you'll not be able to define the same qtree name on different volume was to call:
```
# create qtree qtreename on volume volname
netapp_qtree { 'qtreename': 
  volume=> 'volname',
}

```
New behaviour (able to define qtree with same name on different volume):
```
# create qtree qtreename on volume vol1 and vol2 using volume and qtname
netapp_qtree { 'qtree1': 
  qtname => 'qtreename',
  volume=> 'vol1',
}
netapp_qtree { 'qtree2': 
  qtname => 'qtreename',
  volume=> 'vol2',
}
# create qtree qtreename on volume vol1 and vol2 using title
netapp_qtree { '/vol1/qtreename': }
netapp_qtree { '/vol2/qtreename': }
```